request-manager/spec/node-check.go

```diff
@@ -35,11 +35,15 @@ type ValidCategoryNodeCheck struct{}
 /* 'category: (job | sequence | conditional)' */
 func (check ValidCategoryNodeCheck) CheckNode(sequenceName string, node Node) error {
 	if !node.IsJob() && !node.IsSequence() && !node.IsConditional() {
+		var category string
+		if node.Category != nil {
```

Comment:
You should check if category == nil before the is job/sequence/conditional checks. Else if category is not set, you'll get 2 errors — from `HasCategoryNodeCheck` and from this check.